src/background.js

```diff
@@ -238,35 +246,34 @@ function createTray() {
 async function downloadWalletRelease(releaseUrl, version) {
   status = STATUS.WAIT
   loadUrl(STATUS.WAIT)
-
-  await sleep(5000)
-
+  const file = path.join(SHEIKAH_PATH, FILE_NAME[platform])
   return new Promise((resolve, reject) => {
     axios
       .get(releaseUrl, { responseType: 'stream' })
       .then(async response => {
-        const file = `witnet-release-${arch}-${platform}.tar.gz`
+        const str = progress({
+          length: response.headers['content-length'],
+          time: 100 /* ms */,
+        })
+        str.on('progress', function(progress) {
+          win.webContents.send('progress', progress)
+        })
         const pipeline = util.promisify(stream.pipeline)
         // Promise equivalent for response.data.pipe(writeStream)
-        await pipeline(response.data, fs.createWriteStream(file))
+        await pipeline(response.data, str, fs.createWriteStream(file))
         console.info('witnet release downloaded succesfully')
         console.info('Decompressing release...')
         // Decompress tar.gz file
-        tar.x({
-          file,
-          sync: true,
-        })
-        // TODO: tar is not decompressing correctly if a path is given.
-        // Create these files in ./sheikah
-        fs.copyFileSync('witnet', path.join(SHEIKAH_PATH, WITNET_FILE_NAME))
-        fs.copyFileSync(
-          'witnet.toml',
-          path.join(SHEIKAH_PATH, WITNET_CONFIG_FILE_NAME),
-        )
-        fs.writeFileSync(path.join(SHEIKAH_PATH, VERSION_FILE_NAME), version)
-
+        try {
+          const currentCwd = process.cwd()
+          process.chdir(SHEIKAH_PATH)
+          cp.execSync(`tar -xvf ${file}`)
+          process.chdir(currentCwd)
+        } catch (err) {
+          console.error(err)
+        }
         // Remove the compressed file
-        fs.unlinkSync(file)
+        await sleep(8000)
```

Comment:
Could this be reduced?
```suggestion
        await sleep(4000)
```